frontend/common/src/helpers/poolUtils.ts

```diff
@@ -31,3 +14,35 @@ export const formatClassificationString = ({
 }: formatClassificationStringProps): string => {
   return `${group}-0${level}`;
 };
+
+export const getFullPoolAdvertisementTitle = (
+  intl: IntlShape,
+  poolAdvertisement: Maybe<PoolAdvertisement>,
+): string => {
+  if (poolAdvertisement === null || poolAdvertisement === undefined)
+    return intl.formatMessage({
+      id: "D91nGW",
+      defaultMessage: "Job title not found.",
+      description:
+        "Message shown to user when pool name or classification are not found.",
+    });
+
+  const classification = poolAdvertisement.classifications
+    ? poolAdvertisement.classifications[0]
+    : null;
+
+  let classificationSuffix = ""; // type wrangling the complex type into a string
+  if (classification) {
+    classificationSuffix = formatClassificationString({
+      group: classification?.group,
+      level: classification?.level,
+    });
+  }
+  const genericTitle = getLocalizedName(poolAdvertisement.name, intl);
```

Comment:
I'm pretty sure `poolAdvertisement.name` is considered the "Specific title", and the Classification + Stream is the "Generic" part of the title.